File: apps/api/app/schemas/patient_handoff.py
from pydantic import BaseModel

from app.schemas.patient import PatientOut
from app.schemas.patient_chart import PatientChartSummaryOut
from app.schemas.patient_clinical import (
    PatientCarePlanItemOut,
    PatientEncounterOut,
    PatientLabResultOut,
    PatientMedicationOut,
)
from app.schemas.patient_document import PatientDocumentOut


class PatientCheckoutTaskCreate(BaseModel):
    source_type: str
    source_id: str
    title: str | None = None
    description: str | None = None
    priority: str = "high"
    assigned_to_id: str | None = None


class PatientCheckoutHandoffOut(BaseModel):
    patient: PatientOut
    chart_summary: PatientChartSummaryOut
    documents_needing_review: list[PatientDocumentOut]
    medications_needing_review: list[PatientMedicationOut]
    labs_needing_review: list[PatientLabResultOut]
    care_plan_open_items: list[PatientCarePlanItemOut]
    unsigned_encounters: list[PatientEncounterOut]

File: packages/shared/src/types/patient-handoff.ts
import type { Patient } from './patient';
import type { PatientChartSummary } from './patient-chart';
import type {
  PatientCarePlanItem,
  PatientEncounter,
  PatientLabResult,
  PatientMedication,
} from './patient-clinical';
import type { PatientDocument } from './patient-document';

export interface PatientCheckoutHandoff {
  patient: Patient;
  chart_summary: PatientChartSummary;
  documents_needing_review: PatientDocument[];
  medications_needing_review: PatientMedication[];
  labs_needing_review: PatientLabResult[];
  care_plan_open_items: PatientCarePlanItem[];
  unsigned_encounters: PatientEncounter[];
}
